engine/script/src/script.cpp

```diff
@@ -1850,7 +1850,10 @@ namespace dmScript
             LuaFunctionInfo fi;
             if (dmScript::GetLuaFunctionRefInfo(L, optional_callback_index, &fi))
             {
-                function_source = fi.m_FileName;
+                if (fi.m_FileName)
+                {
+                    function_source = fi.m_FileName;
+                }
```

Comment:
I added one more checking to use `source` from lua_getinfo only if it exists.
There are a few possible values in `source`  one of them is a special `=(tail call)` value which (as I assume) may be  opt out in LuaJit.

Reply:
Is it possible for the filename to be null?

Reply:
`m_FileName` is `source` from `lua_Debug` struct which we get using `lua_getinfo`.

according to doc:
```
source: If the function was defined in a string, then source is that string. If the function was defined in a file, then source starts with a '@' followed by the file name.
```

But reading Lua code I found there is a special branch of logic for tail function:
```
static void info_tailcall (lua_Debug *ar) {
  ar->name = ar->namewhat = "";
  ar->what = "tail";
  ar->lastlinedefined = ar->linedefined = ar->currentline = -1;
  ar->source = "=(tail call)";
  luaO_chunkid(ar->short_src, ar->source, LUA_IDSIZE);
  ar->nups = 0;
}

...
  if (f == NULL) {
    info_tailcall(ar);
    return status;
  }
...
```

Also we know (from attempt to write a Lua profiler) that LuaJit optimise tail calls, so that's why I assume `m_FileName` may be NULL